RevenueCatUI/Templates/Components/Packages/Package/PackageComponentViewModel.swift

```diff
@@ -34,12 +35,32 @@ public class PackageComponentViewModel {
         self.component = component
         self.offering = offering
 
+        self.package = try Self.findPackage(identifier: component.packageID, offering: offering)
+
         self.stackComponentViewModel = try self.component.toStackComponentViewModel(
+            components: self.component.components,
             localizedStrings: localizedStrings,
             offering: offering
         )
     }
 
+    static func findPackage(identifier: String, offering: Offering) throws -> Package {
+        guard let package = offering.package(identifier: identifier) else {
+            Logger.error(Strings.paywall_could_not_find_package(identifier))
+            throw PackageValidationError.missingPackage(
```

Comment:
I think this should never happen with the validations in the `PackageGroupComponentViewModel` right? Maybe we can pass the package insteead of the offering to the constructor here to avoid having to do this validation?

Reply:
Oh good point! I think I added the logic in `PackageGroupComponentViewModel` after this 😅 I will make that change 💪

Reply:
On second thought... going to save this to a follow up PR where we are reworking how every container is a stack because... this is not so easy to do in its current implementation 😬 Right now, view models are _pretty_ independent of each other but that will probably change in the follow up PR. So going to make a task for me to fix this later (later being tomorrow) 💪